presto-orc/src/main/java/com/facebook/presto/orc/StripeReader.java

```diff
@@ -266,8 +275,15 @@ public Stripe readStripe(StripeInformation stripe, AggregatedMemoryContext syste
         // transform streams to OrcInputStream
         ImmutableMap.Builder<StreamId, OrcInputStream> streamsBuilder = ImmutableMap.builder();
         for (Entry<StreamId, OrcDataSourceInput> entry : streamsData.entrySet()) {
-            OrcDataSourceInput sourceInput = entry.getValue();
-            streamsBuilder.put(entry.getKey(), new OrcInputStream(orcDataSource.getId(), sourceInput.getInput(), decompressor, systemMemoryUsage, sourceInput.getRetainedSizeInBytes()));
+            FixedLengthSliceInput input;
+            // TODO: make this configurable
+            if (entry.getKey().getStreamKind() == BLOOM_FILTER || entry.getKey().getStreamKind() == ROW_INDEX) {
```

Comment:
Is this necessary? BTW, this is `isIndexStream()`

Reply:
My understanding is that only `BLOOM_FILTER` and `ROW_INDEX` stand in the critical path (we need to read the actual value for `readStripe`), others could be lazily read.